### PR TITLE
feature(features:main:theme): add theme manager for switch between `dark` and `light` themes

### PR DESCRIPTION
**Pull request by esmaeil-ahmadipour**

### CODE REVIEW

File: lib/src/features/main/theme/theme_data/app_theme_data.dart [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    ),
+    headlineLarge: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w300,
+      fontSize: 36,
+      height: 52 / 36,
+      leadingDistribution: TextLeadingDistribution.even,
+    ),
```
Comment (phoenixit99) [inline lib/src/features/main/theme/theme_data/app_theme_data.dart:62]: Please create a method to reduce code duplication. For ex
 
  static TextStyle _createCommonTextStyle({
    required String fontFamily,
    required Color color,
    required FontStyle fontStyle,
    required FontWeight fontWeight,
    required double fontSize,
    required double height,
    required TextLeadingDistribution leadingDistribution,
  }) {
    return TextStyle(
      fontFamily: fontFamily,
      color: color,
      fontStyle: fontStyle,
      fontWeight: fontWeight,
      fontSize: fontSize,
      height: height,
      leadingDistribution: leadingDistribution,
    );
  }

File: lib/src/features/main/theme/theme_data/app_theme_data.dart [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    headlineMedium: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w400,
+      fontSize: 32,
```
Comment (phoenixit99) [inline lib/src/features/main/theme/theme_data/app_theme_data.dart:68]: Please consider defining the font size as a constant 

  static const fontSizes = FontSizes(
    displaySmall: 16,
    headlineLarge: 36,
    // ... other sizes
  );

File: lib/src/features/main/theme/theme_data/app_theme_data.dart (hunk 1 above)
Comment (esmaeil-ahmadipour) [reply lib/src/features/main/theme/theme_data/app_theme_data.dart:62]: @phoenixit99  ,  Understood. This is a valid point, and I’ll address it in a separate issue at a more appropriate time, aligned with  issue [#34](https://github.com/pactus-project/pactus-gui/issues/34)

File: lib/src/features/main/theme/theme_data/app_theme_data.dart (hunk 2 above)
Comment (esmaeil-ahmadipour) [reply lib/src/features/main/theme/theme_data/app_theme_data.dart:68]: If all the issues are related to text styles, we already have a dedicated issue addressing this.
We can proceed with finishing this merge request.

----
@phoenixit99

Comment (phoenixit99) [reply lib/src/features/main/theme/theme_data/app_theme_data.dart:68]: Ok.got it We can merge request the PR

**Pull request merged**